Cannot find module '../../package' from 'node.js'

- Issue

With the latest version `21.2.0` I get the error `Cannot find module '../../package' from 'node.js'` in all of my tests. I already changed my jest config from `".(ts|tsx)": "<rootDir>/node_modules/ts-jest/preprocessor.js"` to `"^.+\\.tsx?$": "ts-jest"` but the error remains.



- Expected behavior

Tests should run as before.

--- a/AUTHORS
+++ b/AUTHORS
@@ -30,7 +30,6 @@ Justin Bay <jwbay@users.noreply.github.com>
 Kulshekhar Kabra <kulshekhar@users.noreply.github.com>
 Kyle Roach <kroach.work@gmail.com>
 Marshall Bowers <elliott.codes@gmail.com>
-Martijn The <post@martijnthe.nl>
 Matheus Gambati <matheusgambati@gmail.com>
 Maxim Samoilov <samoilowmaxim@gmail.com>
 Mohammad Rajabifard <mo.rajbi@gmail.com>

--- a/README.md
+++ b/README.md
@@ -57,7 +57,7 @@ Modify your project's `package.json` so that the `jest` section looks something 
 {
   "jest": {
     "transform": {
-      "^.+\\.tsx?$": "ts-jest"
+      "^.+\\.tsx?$": "<rootDir>/node_modules/ts-jest/preprocessor.js"
     },
     "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
     "moduleFileExtensions": [
@@ -225,7 +225,7 @@ In `package.json`, inside `jest` section, the `transform` should be like this:
 ```json
 "transform": {
   "^.+\\.jsx?$": "<rootDir>/node_modules/babel-jest",
-  "^.+\\.tsx?$": "ts-jest"
+  "^.+\\.tsx?$": "<rootDir>/node_modules/ts-jest/preprocessor.js"
 }
 ```
 
@@ -236,7 +236,7 @@ Fully completed jest section should look like this:
     "preset": "react-native",
     "transform": {
       "^.+\\.jsx?$": "<rootDir>/node_modules/babel-jest",
-      "^.+\\.tsx?$": "ts-jest"
+      "^.+\\.tsx?$": "<rootDir>/node_modules/ts-jest/preprocessor.js"
     },
     "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
     "moduleFileExtensions": [
@@ -268,7 +268,7 @@ You'll also need to extend your `transform` regex with `html` extension:
 {
   "jest": {
     "transform": {
-      "^.+\\.(tsx?|html)$": "ts-jest"
+      "^.+\\.(tsx?|html)$": "<rootDir>/node_modules/ts-jest/preprocessor.js"
     }
   }
 }

--- a/index.js
+++ b/index.js
@@ -1,0 +1,1 @@
+module.exports = require('./dist/index');

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "ts-jest",
-  "version": "21.1.4",
+  "version": "21.2.1",
   "main": "index.js",
   "types": "./dist/index.d.ts",
   "description": "A preprocessor with sourcemap support to help use Typescript with Jest",

--- a/package.json
+++ b/package.json
@@ -1,8 +1,8 @@
 {
   "name": "ts-jest",
-  "version": "21.2.0",
-  "main": "preprocessor.js",
-  "types": "./dist/preprocessor.d.ts",
+  "version": "21.1.4",
+  "main": "index.js",
+  "types": "./dist/index.d.ts",
   "description": "A preprocessor with sourcemap support to help use Typescript with Jest",
   "scripts": {
     "build": "tsc -p .",
@@ -78,7 +78,7 @@
     "yargs": "^10.0.3"
   },
   "peerDependencies": {
-    "jest": "^21.1.0 || ^21.3.0-alpha.1 || ^22.0.0-alpha.1",
+    "jest": "^21.1.0 || ^21.1.0-alpha.1 || ^22.0.0-alpha.1",
     "typescript": "2.x"
   },
   "devDependencies": {

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,6 +1,7 @@
 import * as sourceMapSupport from 'source-map-support';
 import { defaultRetrieveFileHandler } from './default-retrieve-file-handler';
 
+export { transpileIfTypescript } from './transpile-if-ts';
 export function install() {
   const options: sourceMapSupport.Options = {};
 

--- a/src/preprocessor.ts
+++ b/src/preprocessor.ts
@@ -6,8 +6,6 @@ import { JestConfig, Path, TransformOptions } from './jest-types';
 import { getPostProcessHook } from './postprocess';
 import { getTSConfig, getTSJestConfig } from './utils';
 
-export { install } from './install';
-
 export function process(
   src: string,
   path: Path,

--- a/tests/__tests__/transpile-if-ts.spec.ts
+++ b/tests/__tests__/transpile-if-ts.spec.ts
@@ -1,4 +1,4 @@
-import { transpileIfTypescript } from '../../src/transpile-if-ts';
+import { transpileIfTypescript } from '../../src';
 
 describe('transpileIfTypescript', () => {
   it('should ignore anything non-TS', () => {

--- a/tests/jestconfig-test/jest.json
+++ b/tests/jestconfig-test/jest.json
@@ -1,7 +1,7 @@
 {
   "rootDir": "./",
   "transform": {
-    "^.+\\.tsx?$": "ts-jest"
+    "^.+\\.tsx?$": "<rootDir>/node_modules/ts-jest/preprocessor.js"
   },
   "mapCoverage": true,
   "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",

